apiserver/observer/requestlogger.go

```diff
@@ -0,0 +1,190 @@
+// Copyright 2016 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package observer
+
+import (
+	"context"
+	"net/http"
+	"time"
+
+	"github.com/juju/clock"
+	"github.com/juju/names/v6"
+
+	"github.com/juju/juju/core/logger"
+	"github.com/juju/juju/core/model"
+	"github.com/juju/juju/rpc"
+	"github.com/juju/juju/rpc/jsoncodec"
+)
+
+// RequestLoggerConfig provides information needed for a
+// RequestLogger to operate correctly.
+type RequestLoggerConfig struct {
+	// Clock is the clock to use for all time operations on this type.
+	Clock clock.Clock
+
+	// Logger is the log to use to write log statements.
+	Logger logger.Logger
+}
+
+// RequestLogger serves as a sink for API server requests and
+// responses.
+type RequestLogger struct {
+	BaseObserver
+
+	clock      clock.Clock
+	logger     logger.Logger
+	connLogger logger.Logger
+	pingLogger logger.Logger
+
+	id                 uint64
+	websocketConnected time.Time
+}
```

Comment:
This feels weird. Why do we need an observer, which simply logs out login, logout requests. Does it even use the BaseObserver?

Reply:
Yes, it uses it for the agent tag.